packages/plugin/src/plugin/data.ts

```diff
@@ -249,6 +262,37 @@ function matchesEntryPoint(
 	);
 }
 
+function modMatchesEntryPoint(
+	mod: JSONOutput.DeclarationReflection,
+	entry: PackageEntryConfig,
+	cfg: {
+		packagePath: string;
+		isSinglePackage: boolean;
+	},
+) {
+	const relModSources = mod.sources ?? [];
+	const relModSourceFile = relModSources.find((sf) => !!sf.fileName)?.fileName ?? '';
+	const relEntryPoint = joinUrl(cfg.packagePath, entry.path);
+	const isUsingDeepImports = !entry.path.match(/\.tsx?$/);
+
+	// Monorepos of 1 package don't have sources, so use the child sources
+	if (!relModSourceFile) {
+		const relSourceFiles = buildSourceFileNameMap(mod.children ?? []);
+		const relEntryPointInSourceFiles = !!relSourceFiles[relEntryPoint];
+		if (relEntryPointInSourceFiles) {
+			return matchesEntryPoint(relEntryPoint, relEntryPoint, {
```

Comment:
wip: I assumed it should just run the matching logic due to the `deep` or `single` flags affecting whether the source `fileName` is valid for the `relEntryPoint`. Since it's an exact match, I was wondering if it could just return `true` here.

Reply:
lets keep it just to be 100% sure its valid